### ISSUE TITLE
quantized_scope does not return binary weights

### ISSUE THREAD

**Issue by ddkang** [larq/larq#792]
### Describe the bug
quantized_scope does not return binary weights

### To Reproduce
Running the following code produces all float weights (and manually inspecting them confirms they are not binary values):
```
import larq_zoo as lqz
import larq as lq

model = lqz.literature.XNORNet(weights='imagenet')

with lq.context.quantized_scope(True):
    weights = model.get_weights()
    for weight in weights:
        print(weight.dtype)
```

### Expected behavior
Binary weights

### Environment

TensorFlow version: 2.9.1
Larq version: 0.12.2

**Comment by ddkang** [comment]
I also ran this code snippet (with the imagenet weights): https://spectrum.chat/larq/general/saving-binary-weights~c139979c-d9bc-4edc-80d7-687155898e7d?m=MTU5NzMxMDczMTg2NQ==

It still outputs floating point numbers for me:
```
import larq as lq
import larq_zoo as lqz
import tensorflow as tf

# Load model and save binary weights
model = lqz.sota.QuickNet(weights='imagenet')
with lq.context.quantized_scope(True):
    model.save('TempSavedModel')

# Reload model and print weights
model = tf.keras.models.load_model('TempSavedModel')
for layer in model.layers:
    if isinstance(layer, lq.layers.QuantConv2D):
        print(layer.weights)
        break
```

Output:
```
[<tf.Variable 'quant_conv2d/kernel:0' shape=(3, 3, 3, 16) dtype=float32, numpy=
array([[[[-0.76475644,  0.24302398,  1.4282253 , -2.6767862 ,
           1.9444721 ,  0.43911752,  0.03137949, -3.1915727 ,
          -0.31189498,  1.3543414 ,  0.7486021 , -2.2474093 ,
          -0.19764178, -1.1965723 , -0.63690233, -0.07212169],
```

**Comment by Tombana** [comment]
I believe you're looking at the weights of the first layer of the network. The first layer in these binary networks is usually *not* binarized.

**Comment by ddkang** [comment]
If I run this:
```
import larq as lq
import larq_zoo as lqz
import tensorflow as tf

# Load model and save binary weights
model = lqz.sota.QuickNet(weights='imagenet')
with lq.context.quantized_scope(True):
    model.save('TempSavedModel')

# Reload model and print weights
model = tf.keras.models.load_model('TempSavedModel')
for layer in model.layers:
    if isinstance(layer, lq.layers.QuantConv2D):
        print(layer.weights)
```
(without the break)
I still get that every set of weights is floating point:
```
<QuantizedVariable 'quant_conv2d_15/kernel:0' shape=(3, 3, 256, 256) dtype=float32 quantizer=SteSign precision=1 numpy=
array([[[[-1.2862252e-02,  4.6198699e-01,  1.2496794e+00, ...,
          -1.2231784e+00,  1.0684882e+00, -8.1849754e-01],
```
Do you know if it's possible to get the _binary_ weights?

**Comment by jneeven** [comment]
Hi, this is a duplicate of #553. Please see the explanation there.
In summary: if you want the binary weights, you can use
```python
with lq.context.quantized_scope(True):
    model.set_weights(model.get_weights())  # remove latent weights
 ```

But note that this will break training if you then try to continue training this model, since the latent weights have been destroyed. So make sure this is actually what you want.

**Comment by ddkang** [comment]
The code snippet above works for QuickNet but does not work for XNORNet:
 ```
 import larq as lq
import larq_zoo as lqz
import tensorflow as tf

# Load model and save binary weights
model = lqz.literature.XNORNet(weights='imagenet')

with lq.context.quantized_scope(True):
    model.set_weights(model.get_weights())
    model.save('TempSavedModel')

# Reload model and print weights
model = tf.keras.models.load_model('TempSavedModel')
for layer in model.layers:
    print(type(layer))
    if isinstance(layer, lq.layers.QuantConv2D):
        print(layer.weights[0])
```

**Comment by Tombana** [comment]
The weights are binary, they're of the form `-a, a`, but the magnitude `|a|` is different for each output channel. That magnitude can be fused into the batchnorm multiplier to make the weights `-1, +1` again.
